presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -991,6 +992,25 @@ protected Scope visitExplain(Explain node, Optional<Scope> scope)
         @Override
         protected Scope visitQuery(Query node, Optional<Scope> scope)
         {
+            Statement statement = analysis.getStatement();
+
+            if (statement instanceof Query && !analysis.isRewritten()) {
+                QualifiedName baseTableName = (QualifiedName) new ExtractTableNameVisitor<>().process(node);
```

Comment:
create a method to validate the query.
ifOptimizeQueryWithMaterializeView(node)